src/main/scala/ayai/systems/ai/QuestGenerationSystem.scala

```diff
@@ -22,55 +24,73 @@ object QuestGenerationSystem {
-            1,                    // id
-            "hello",              // title
-            "do a thing",         // description
-            12,                   // recommended level
-            objectives.toList     // objectives
-          )
-
-          // add a new quest to the quest bag of the initiator!
-          genQuest.initiator.getComponent(classOf[QuestBag]).asInstanceOf[QuestBag].addQuest( questComponent )
-          // the gamestate needs to keep a list of our quests?
-          //networkSystem.actorSelection("user/QuestMap") ! AddQuest("QUEST"+questData.id, questComponent)
-
-          // lastly, remove the generation request. We don't need it anymore.
-          e.removeComponent( classOf[GenerateQuest] )
+          // the next thing we'll need is the memory of the quest-giver. we'll use their memories to figure out what the
+          // quest should target.
+          genQuest.initiator.getComponent(classOf[Memory]) match {
+            case Some(initiatorMemory: Memory) =>
+
+              //val recipientHistory = genQuest.recipient.getComponent(classOf[QuestHistory])
+
+              // create an objective list
+              var objectives = new ListBuffer[Objective]()
+
+              // next, we need to populate our quest with objectives. To do this, we'll find the most "significant" thing
+              // in an NPC's memory and build objectives based on that.
+              var mostSignificantMemory: MemoryContents = null
+              for (memory <- initiatorMemory.entitiesRemembered) {
+                if (math.abs(memory.relationship) > math.abs(mostSignificantMemory.relationship)) {
+                  mostSignificantMemory = memory
+                }
+              }
+
+              // next, we can decide whether that thing is "negative" or "positive". If it's negative, the NPC will want that
+              // thing to be killed or something. If it's positive, maybe you should bring it to them.
+              if ( mostSignificantMemory != null ) {
+                if ( mostSignificantMemory.relationship > 0 ) {
+                  // positive relationship quests
+                  objectives += new FetchObjective( "Fetch A thing", "Get this", "Bring it here" )
+                }else{
+                  // negative relationship quests.
+                  objectives += new KillObjective( "Get Entity Name from Passed ID Here", 0, 10 )
+                }
+              }else{
+                // this character has literally nothing on his mind... Umm... I guess return null?
+                return null
+              }
+
+              // so... yeah, sorry about this! Each quest needs an integer ID, preferably one that doesn't conflict with
+              // any others. Time's pretty much contiguous, so as long as you don't try and run this game for more than 24 days,
+              // at a time, we should be good! Fix this as soon as possible.
+              val id = new Date().getTime().toInt
```

Comment:
I think the program failing randomly would be more annoying to debug than the program failing after 24 days.

If you want to fix this, one way is to actually check for uniqueness of the generated IDs. You could keep a `var usedQuestIds: Set` around. But that would be redundant, since the quests already exist. Instead, you could look up all existing quests, and look at their IDs to see if the one you generated is taken. If it is, try that ID + 1.

Another way to fix it would be to refactor `Quest` so that its `id` is not an `Int`, but a `String`. Then generate a UUID and convert it to a string like we discussed on GroupMe.